### ISSUE TITLE
has-body and response/status don't play nicely together

### ISSUE THREAD

**Issue by paulbutcher** [noir-clojure/noir#84]
The following page definition and associated test work just fine:

``` clojure
(defpage "/welcome1" []
  (response/json "hello"))

(deftest test-welcome1
  (-> (send-request "/welcome1")
      (has-body "\"hello\"")))
```

But the following:

``` clojure
(defpage "/welcome2" []
  (response/status 201 (response/json "hello")))

(deftest test-welcome2
  (-> (send-request "/welcome2")
      (has-body "\"hello\"")))
```

Fails with:

```
FAIL in (test-welcome2) (test.clj:38)
expected: (= cont (get resp :body))
  actual: (not (= "\"hello\"" {:headers {"Content-Type" "application/json"}, :body "\"hello\""}))
```

I can work around this:

``` clojure
(deftest test-welcome2-fixed
  (let [resp (send-request "/welcome2")]
    (has-body (:body resp) "\"hello\"")))
```

but that's not exactly very nice.

**Comment by paulbutcher** [comment]
Sorry - I've just realised that the problem is much more than the test failing. I assumed that `status` could be used to set the status for any response type. But clearly that's not the case.

This seems to be the way to change the status for a non-text body?

``` clojure
(defpage "/welcome3" []
  (merge (response/json "hello") {:status 201}))
```

Any reason why `status` couldn't be modified to work in the way I initially assumed it should?

**Comment by ibdknox** [comment]
Nope, sounds like it would be a welcomed, and simple change :)

**Comment by paulbutcher** [comment]
See #85

**Comment by ibdknox** [comment]
fixed in 1.3.0-beta1
